File: Cpp/Day 8/inheritance/Assi1/person.cpp
#include"person.h"
Person::Person()
{

}
    Person::Person(const char *ch,int d,int m,int y)
    {
        this->nm=cString(ch);
        bday=Date(d,m,y);
    }
    void Person::display()
    {
        nm.display();
        bday.display();
    }

File: Java lab/polyoffice/src/polyoffice/util/Person.java
package polyoffice.util;

import java.io.Serializable;

public class Person implements Serializable{
	private String name;
	private Date bdate;
	public Person(String name, Date bdate) {
		this.name = name;
		this.bdate = bdate;
	}
	public Person(String name, int dd,int mm,int yy) {
		this.name = name;
		this.bdate = new Date(dd,mm,yy);
	}
	public Person() {
		name="Not given";
		bdate=new Date();
	}
	public String toString()
	{
		return "Name: "+name+"\nBirthdate: "+bdate.tostring();
	}
	public void display()
	{
		System.out.println("Name: "+name);
		bdate.showdate();
	}

}
